core/src/main/kotlin/fr/sncf/osrd/sim_infra_adapter/RawInfraFromRjsAdapter.kt

```diff
@@ -535,8 +536,60 @@ fun adaptRawInfra(infra: SignalingInfra, rjsInfra: RJSInfra): SimInfraAdapter {
         }
     }
 
-    // TODO: remove this stitching between new way of loading track-section info and previous way of
-    // loading network
+    // parse nodes
+    val switchTypeMap =
+        listOf(rjsInfra.switchTypes, RJSSwitchType.BUILTIN_NODE_TYPES_LIST)
+            .flatten()
+            .map { switchType -> Pair(switchType.id, switchType) }
+            .toMap()
```

Comment:
Fancy suggestion from intellij

```suggestion
            .associateBy { switchType -> switchType.id }
```

Reply:
Done in https://github.com/osrd-project/osrd/commit/c4d0a6a9f7a45fa3697f8a392413181cd6cc6e5d